lib/Doctrine/DBAL/Platforms/Keywords/SQLiteKeywords.php

```diff
@@ -15,6 +17,14 @@ public function getName()
         return 'SQLite';
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isKeyword($word)
+    {
+        return parent::isKeyword($word) || is_numeric($word);
```

Comment:
Is this an alternative to #4246? `5` doesn't look like an SQLite keyword, so this change doesn't look right. As pointed out earlier, the problem is that an invalid key name is generated in the first place, not that it's not escaped.

Reply:
Yes and no, no one prevents a user to use numbers as FK names so it can break for them in the future. The reason its here is to not clutter \Doctrine\DBAL\Schema\AbstractAsset::getQuotedName with some numerical checks, esp. as i don't know how other databases handle this and at the end this was the fastest (aka least code change/restructuring) change

Reply:
We're currently reevaluating the approach to quoting names leaning towards the user being responsible for quoting since the existing magic is unreliable anyways.

> no one prevents a user to use numbers as FK names so it can break for them in the future

I believe it never worked, so it's a good enough preventing measure.

Regardless of that, it doesn't look like a bugfix so `2.10.x` is not the right place for this change.

Reply:
At least it would be the smallest hotfix for `2.10.x` as its currently broken for anyone changing tables so i would consider it a bugfix? But i totally agree with the point that it definitely does not feel right to add it there (but again, its the "minimum change"/the change that i am currently comfortable to do in this project :grin:)